### PR TITLE
Bytepatch Fixes

### PR DESCRIPTION
**Pull request by BenCat07**
Make bullet traces only init when used, and make killfeed color be off by default, also rework bytepatch to do all the searching logic at patch time instead of construction.

### CODE REVIEW

File: src/hacks/KillfeedColor.cpp [hunk 1]
```diff
@@ -6,7 +6,7 @@
 #if !ENFORCE_STREAM_SAFETY
 namespace hacks::tf2::killfeed
 {
-static settings::Boolean enable{ "visual.killfeedcolor.enable", "true" };
+static settings::Boolean enable{ "visual.killfeedcolor.enable", "false" };
```
Comment (TotallyNotElite) [inline src/hacks/KillfeedColor.cpp:9]: Huh?

Comment (BenCat07) [reply src/hacks/KillfeedColor.cpp:9]: Don't enable by default, idk why it was, if you want i can change it back to true

Comment (TotallyNotElite) [reply src/hacks/KillfeedColor.cpp:9]: idk i kinda like it that way

Comment (TotallyNotElite) [reply src/hacks/KillfeedColor.cpp:9]: And cause it doesn't impact gameplay, I'm fine with it being true by default

Comment (TotallyNotElite) [inline src/hacks/KillfeedColor.cpp:9]:
```suggestion
static settings::Boolean enable{ "visual.killfeedcolor.enable", "true" };
```

**Pull request merged**